pkg/controller/publisher.go

```diff
@@ -160,46 +158,69 @@ func (s *SCSIPublisher) addLUNIDToPublishContext(
 	publishContext map[string]string,
 	mapping gopowerstore.HostVolumeMapping,
 	volume gopowerstore.Volume,
+	isRemote bool,
 ) {
-	publishContext[common.PublishContextDeviceWWN] = strings.TrimPrefix(volume.Wwn, common.WWNPrefix)
-	publishContext[common.PublishContextLUNAddress] = strconv.FormatInt(mapping.LogicalUnitNumber, 10)
+	if !isRemote {
+		publishContext[common.PublishContextDeviceWWN] = strings.TrimPrefix(volume.Wwn, common.WWNPrefix)
+		publishContext[common.PublishContextLUNAddress] = strconv.FormatInt(mapping.LogicalUnitNumber, 10)
+	} else {
+		publishContext[common.PublishContextRemoteDeviceWWN] = strings.TrimPrefix(volume.Wwn, common.WWNPrefix)
+		publishContext[common.PublishContextRemoteLUNAddress] = strconv.FormatInt(mapping.LogicalUnitNumber, 10)
+	}
 }
 
 func (s *SCSIPublisher) addTargetsInfoToPublishContext(
-	publishContext map[string]string, volumeApplianceID string, client gopowerstore.Client,
+	publishContext map[string]string, volumeApplianceID string, client gopowerstore.Client, isRemote bool,
 ) error {
+	iscsiPortalsKey := common.PublishContextISCSIPortalsPrefix
+	iscsiTargetsKey := common.PublishContextISCSITargetsPrefix
+	fcWwpnKey := common.PublishContextFCWWPNPrefix
+	nvmeFcPortalsKey := common.PublishContextNVMEFCPortalsPrefix
+	nvmeFcTargetsKey := common.PublishContextNVMEFCTargetsPrefix
+	nvmeTCPPortalsKey := common.PublishContextNVMETCPPortalsPrefix
+	nvmeTCPTargetsKey := common.PublishContextNVMETCPTargetsPrefix
```

Comment:
Are these being logged somewhere in case we need to troubleshoot an issue?

Reply:
Yes, but not directly. First set of logs for local volume and if remote ID is present, second set of logs for remote volume.